### ISSUE TITLE
libgit2 doesn't handle .gitignore/exclude patterns with a trailing slash and white space

### ISSUE THREAD

**Issue by segevfiner** [libgit2/libgit2#4610]
How "fun"... 😔 

### Reproduction steps
1. Execute:
    ```sh
    git init test-repo && cd test-repo
    mkdir spam
    touch spam/ham.c
    echo "spam/ " > .gitignore
    ```
2. `git status --ignored` shows that "spam/" is ignored.
3. With `pygit2`:
    ```py
    >>> from pygit2 import Repository
    >>> r = Repository('.')
    >>> r.path_is_ignored('spam')
    False
    >>> r.path_is_ignored('spam/ham.c')
    False
    ```

### Expected behavior
The spam directory should be ignored.

### Actual behavior
It isn't.

### Version of libgit2 (release number or SHA1)
Tested both v0.27.0 and master at d9007dc8520942763a21e4058dfa8f159f3e3c56.

### Operating system(s) tested
Ubuntu 16.04 amd64
Windows 10.0.16299.334 x64

### Related
https://github.com/libgit2/libgit2/issues/4309
https://github.com/libgit2/libgit2/commit/5cb6a2c946a071d43be3cdabf2f6d4417a4be694

cc @pks-t @novalis

**Comment by pks-t** [comment]
Thanks for the report. I can reproduce the error and am working on a fix

**Comment by pks-t** [comment]
Please verify that #4614 fixes your issue

**Comment by ethomson** [comment]
Thanks for reporting this.  Fixed via #4614.
